### ISSUE TITLE
`check` in `select` results in runtime panic

### ISSUE THREAD

**Issue by KavinduZoysa** [ballerina-platform/ballerina-lang#33473]
**Description:**
$subject. As mentioned in this commit,
https://github.com/ballerina-platform/nballerina/commit/c129ef3334fe48279b0621bab7091c8ba889732b#diff-55d6c1329641c5efbc3b2e8501bb7892094329f3c3744e2eb221950976b0d3ffR148

If we replace foreach with from/select, it will print the backtrace as shown below. We can get this by running [this example](https://github.com/ballerina-platform/nballerina/blob/main/compiler/testSuite/future/record1-uv.bal), against nballerina 

```
error: type not implemented as member type {"location":{"filename":"test.bal","startPos":[4,4],"endPos":null},"functionName":null}
	at wso2.nballerina.err.0:unimplemented(err.bal:64)
	   wso2.nballerina.front.0:resolveMemberTypeDesc(resolveTypes.bal:258)
	   wso2.nballerina.front.0:$streamLambda$_21(resolveTypes.bal:153)
	   ballerina.lang.query.0._SelectFunction:process(types.bal:590)
	   ballerina.lang.query.0:$annon$method$delegate$0(types.bal:91)
	   ballerina.lang.__internal.0:invokeAsExternal(stream-utils.bal:138)
	   ballerina.lang.query.0._StreamPipeline:next(types.bal:91)
	   ballerina.lang.query.0.IterHelper:next(types.bal:723)
	   ballerina.lang.stream.0:next(stream.bal:67)
	   ballerina.lang.query.0:toArray(helpers.bal:88)
	   wso2.nballerina.front.0:resolveTypeDesc(resolveTypes.bal:153)
	   wso2.nballerina.front.0:resolveTypeDefn(resolveTypes.bal:69)
	   wso2.nballerina.front.0:resolveTypes(resolveTypes.bal:29)
	   wso2.nballerina.front.0:resolveModule(front.bal:140)
	   wso2.nballerina.0:processModule(compile.bal:90)
	   wso2.nballerina.0:compileBalFile(compile.bal:76)
	   wso2.nballerina.0:main(main.bal:51)
```
**Steps to reproduce:**
Still I did not find a simple example to reproduce this. I will try to find a simple example

**Affected Versions:**
swan-lake beta3

**Comment by jclark** [comment]
Another example here: https://github.com/ballerina-platform/nballerina/commit/569f8fc338726be6fb6f52e956a66467934cba2a

**Comment by gimantha** [comment]
Simplest code I could write to reproduce this issue, 

```ballerina
type Semantic distinct error;
type Unimplemented distinct error;
type ResolveTypeError Semantic|Unimplemented;

public function main() returns error? {
    any|error x = eee();
    if x is error {
        return;
    }
}

public function eee() returns any|ResolveTypeError {
    int[] td = [1, 2, 3, 4, 5, 8, 9];
    int[] params = [];
    params = from var x in td select check foo(x);
    //foreach var x in td {
    //    params.push(check foo(x));
    //}
    return params;
}

function foo(int td) returns int|ResolveTypeError {
    if td > 5 {
        return td;
    }
    return error Unimplemented("Apoi");
}
```

If I used `error` for all the places where `ResolveTypeError` is, then it does not panic.

**Comment by jclark** [comment]
Ahh, it's the error typing that is causing the problems. Thanks @gimantha for pursuing this.

**Comment by lochana-chathura** [comment]
Fixed in #35344
